cloudinit/net/dhcp.py

```diff
@@ -679,6 +703,38 @@ def dhcp_discovery(
             )
             raise NoDHCPLeaseError from error
 
+    @staticmethod
+    def parse_unknown_options_from_packet(
+        data: bytes, number: int
+    ) -> Optional[bytes]:
+        """get a specific option from a binary lease file
+
+        @param data: Binary lease data
+        @param number: Option number to return
+        @return: the option (bytes) or None
+        """
+        # comes right after the magic cookie
```

Comment:
Can we reference the rfc that defined magic cookie at index 240?

Cross referencing other rfcs I see good mention of magic cookie as the first "marker" in the vend field but nothing specifically defining that index/position of the field.
https://www.ietf.org/rfc/rfc1497.txt and https://www.ietf.org/rfc/rfc2132.txt make reference to it, but not it's index.

I know this implementation is correct as I've poked around and tested everything from option 1,3,6 and 51 and seeing the expected netmask GW etc.

When looking through the following https://www.ietf.org/rfc/rfc951.txt I can see the bootstrap protocol which looks to define the `vend` field which looks to be nearly the same byte index 240 if I add up all the preceding fields.

Reply:
Maybe we just reference the struct here?https://github.com/NetworkConfiguration/dhcpcd/blob/master/src/dhcp.h#L167

Reply:
> I can see the bootstrap protocol which looks to define the vend field which looks to be nearly the same byte index 240 if I add up all the preceding fields.

Yep, the vend field starts on 236, so if you add the 4 bytes that the magic cookie takes up that's where we get index 240. I've updated the comments to make that easier to discover for future readers of this code.